include/envoy/upstream/upstream.h

```diff
@@ -732,6 +732,11 @@ class ClusterInfo {
    */
   virtual const absl::optional<std::chrono::milliseconds> idleTimeout() const PURE;
 
+  /**
+   * @return the idle timeout for upstream connection pools.
+   */
+  virtual std::chrono::milliseconds poolIdleTimeout() const PURE;
```

Comment:
nit: I would return absl::optional here as opposed to 0 to make it more clear when there is no timeout.